docs/sources/flow/reference/components/otelcol.receiver.aws_firehose.md

```diff
@@ -0,0 +1,164 @@
+`max_request_body_size` | `string` | Maximum request body size the HTTP server will allow. No limit when unset. | | no
+`include_metadata` | `boolean` | Propagate incoming connection metadata to downstream consumers. | | no
+
+`access_key` can be set when creating or updating the delivery stream. See the [AWS Firehose documentation](https://docs.aws.amazon.com/firehose/latest/dev/create-destination.html#create-destination-http) for more details.
+The supported values for `record_type`  are:
+* `cwmetrics`: The record type for the CloudWatch metric stream. Expects the format for the records to be JSON. See the [CloudWatch documentation][cloudwatch-metric-streams] for details.
+[cloudwatch-metric-streams]: https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-Metric-Streams.html
+
+## Blocks
+
+The following blocks are supported inside the definition of
+`otelcol.receiver.awsfirehose`:
+
+Hierarchy | Block | Description | Required
+--------- | ----- | ----------- | --------
+tls | [tls][] | Configures TLS for the HTTP server. | no
+cors | [cors][] | Configures CORS for the HTTP server. | no
+debug_metrics | [debug_metrics][] | Configures the metrics that this component generates to monitor its state. | no
+output | [output][] | Configures where to send received traces. | yes
+
+The `>` symbol indicates deeper levels of nesting. For example, `grpc > tls`
+refers to a `tls` block defined inside a `grpc` block.
+
+[tls]: #tls-block
+[cors]: #cors-block
+[debug_metrics]: #debug_metrics-block
+[output]: #output-block
+
+### tls block
+
+The `tls` block configures TLS settings used for a server. If the `tls` block
+isn't provided, TLS won't be used for connections to the server.
+
+{{< docs/shared lookup="flow/reference/components/otelcol-tls-config-block.md" source="agent" version="<AGENT_VERSION>" >}}
+
+### cors block
+
+The `cors` block configures CORS settings for an HTTP server.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`allowed_origins` | `list(string)` | Allowed values for the `Origin` header. | | no
+`allowed_headers` | `list(string)` | Accepted headers from CORS requests. | `["X-Requested-With"]` | no
+`max_age` | `number` | Configures the `Access-Control-Max-Age` response header. | | no
+
+The `allowed_headers` argument specifies which headers are acceptable from a
+CORS request. The following headers are always implicitly allowed:
+
+* `Accept`
+* `Accept-Language`
+* `Content-Type`
+* `Content-Language`
+
+If `allowed_headers` includes `"*"`, all headers are permitted.
+
+### debug_metrics block
+
+{{< docs/shared lookup="flow/reference/components/otelcol-debug-metrics-block.md" source="agent" version="<AGENT_VERSION>" >}}
+
+### output block
+
+{{< docs/shared lookup="flow/reference/components/output-block.md" source="agent" version="<AGENT_VERSION>" >}}
```

Comment:
```suggestion
{{< docs/shared lookup="flow/reference/components/output-block-metrics.md" source="agent" version="<AGENT_VERSION>" >}}
```

`output-block.md` lists metrics, logs, and traces. But this component can only output metrics.

Reply:
Yes, but on the other hand this is the shared export of all otelcol receivers and we don't have a way to limit to specific signals only.

Reply:
Yes, I think we should hide them. It will only confuse the users. I think the code should be more restrictive in the future too, and components should fail if they are assigned the wrong output.